### ISSUE TITLE
BUG: PHP Unit Tests are failing on PHP 8

### ISSUE THREAD

**Issue by burhandodhy** [10up/ElasticPress#3049]
**Describe the bug**

PHP Unit tests are failing on PHP 8. Only 5 tests have the issue. 

```
There were 5 errors:

1) ElasticPressTest\TestSearchAlgorithm::testFilters
ParseError: syntax error, unexpected token "match", expecting variable

/var/www/html/wp-content/plugins/elasticpres/tests/php/TestSearchAlgorithm.php:29

2) ElasticPressTest\TestSearchAlgorithm::testLegacyFilters
ParseError: syntax error, unexpected token "match", expecting variable

/var/www/html/wp-content/plugins/elasticpres/tests/php/TestSearchAlgorithm.php:29

3) ElasticPressTest\TestSearchAlgorithms::testRegisterAndGetSearchAlgorithms
ParseError: syntax error, unexpected token "match", expecting variable

/var/www/html/wp-content/plugins/elasticpres/tests/php/TestSearchAlgorithms.php:28

4) ElasticPressTest\TestFacets::testFacetTypeRegistration
ParseError: syntax error, unexpected token "match", expecting variable

/var/www/html/wp-content/plugins/elasticpres/tests/php/features/TestFacet.php:24

5) ElasticPressTest\TestWeighting::testHandleSave
ParseError: syntax error, unexpected token "match", expecting variable

/var/www/html/wp-content/plugins/elasticpres/tests/php/features/TestWeighting.php:251
```

**Steps to Reproduce**

1. Set PHP version to 8
2. Run unit tests.

**Expected behavior**
All tests should pass on PHP 8

**Comment by felipeelia** [comment]
Once we have addressed this one, let's make sure we bump the PHP version used in the GitHub Action too.

**Comment by burhandodhy** [comment]
@felipeelia phpunit 7.5.20 doesn't have support for PHP 8. Should I bump the version?

**Comment by felipeelia** [comment]
Yeah, that won't be simple though. There is an issue for that already: https://github.com/10up/ElasticPress/issues/2459

**Comment by felipeelia** [comment]
Closed by #3073
